lib/Widgets/DefaultWidget.php

```diff
@@ -111,32 +114,36 @@ public function loadWidget(IWidgetConfig $settings) {
 	 */
 	public function requestWidget(IWidgetRequest $request) {
 		if ($request->getRequest() === 'getWeather') {
-
-			$app = new Application();
-			$container = $app->getContainer();
-			$weatherController = $container->query('OCA\Weather\Controller\WeatherController');
-			$cityController = $container->query('OCA\Weather\Controller\CityController');
-			$settingsController = $container->query('OCA\Weather\Controller\SettingsController');
-
-			$allCities = json_decode($cityController->getAll()->render(), true);
-
-			$homeCityId = $allCities['home'];
-			$firstCity = array_filter(
-				$allCities['cities'],
-				function($city) use ($homeCityId) { 
-					return $city['id'] === $homeCityId; 
-				}
-			)[0]['name'];
-	
-			$result = json_decode($weatherController->get($firstCity)->render(), true);
-			$metric = json_decode($settingsController->metricGet()->render(), true)['metric'];
-
-			$request->addResult('location', $firstCity);
-			$request->addResult('temperature', $result['main']['temp']);
-			$request->addResult('metric', $metric);
-			$request->addResult('weather', $result['weather'][0]['description']);
-			$request->addResult('humidity', $result['main']['humidity']);
-			$request->addResult('wind', $result['wind']['speed']);
+			try {
```

Comment:
please don't catch on all the block, only required block

Reply:
Quite right. 
As there were only two points of failure they could be checked separately – thus there isn't even need for a try/catch.